sotodlib/calibration/calibration_tod_analysis.py

```diff
@@ -0,0 +1,749 @@
+            os.makedirs(out_folder)
+        full_df.to_hdf(opj(out_folder, h_name+'.h5'),key='parameter_table', mode='w')
+
+        t2 = time.time()
+        print('Time to run fittings for a full wafer is {} seconds'.format(t2-t1))
+
+        ## check why the mapmaker was working only with context and fix this issue
+        if map_make and f_format=='context' and n_modes is not None:
+            beamsize = np.degrees(init_params[2])/2 
+            make_maps(path, outdir, obs_id, sso_name, beamsize, n_modes)
+
+
+
+def main():
+    
+    parser = ap.ArgumentParser(formatter_class=\
+    ap.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--indir', action='store', dest='indir',
+    help='The location of the simulated files.')
+    parser.add_argument('--outdir', action='store', dest='outdir',
+    help='The location for the .h5 ouput files to be stored.')
+    parser.add_argument('--f_format', action='store', dest='f_format',
+    help='The format of the simulations [g3, h5, context].')
+    parser.add_argument('--tele', action='store', dest='tele',
+    help='The telescope name [LAT, SAT].')
+    parser.add_argument('--tube', action='store', dest='tube',
+    help='The tube name [c1, i6, o6, ...] for LAT, [SAT1, SAT2, ...] for SAT.')
+    parser.add_argument('--band', action='store', dest='band', 
+    help='Frequency band [30,40,90,150,230,290].')
+    parser.add_argument('--wafer', action='store', dest='wafer',
+    help='The wafer name [wafers number increases from the center and \
+    anti-cloackwise] across the focalplane.')
+    parser.add_argument('--sso_name', action='store', dest='sso_name',
+    help='Calibration source (astrophysical only for the time being).')
+    parser.add_argument('--year', action='store', dest='year',
+    help='The year of the calibration observation.')
+    parser.add_argument('--month', action='store', dest='month',
+    help='The month of the calibration observation.')
+    parser.add_argument('--day', action='store', dest='day',
+    help='The day of the calibration observation')
+    ## The telescope can be defined from the wafer name
+    ## No need for extra parser argument --fix later
+
+    parser.add_argument('--highpass', action='store_true', dest='highpass',
+    help='If True, use highpass butterworth filters (especiall useful for SATs).')
+    parser.add_argument('--cutoff', action='store', dest='cutoff', default=None,
+    help='The cutoff frequency to be used in the filtering.', type=float)
+    #parser.add_argument('--sample_rate', action='store', dest='sample_rate', default=None, 
+    #help='Sampling rate of the observation.', type=int)
+    parser.add_argument('--map_make', action='store_true', dest='map_make',
+    help='Make planet maps of the simulations fitted.')
+    parser.add_argument('--n_modes', action='store', dest='n_modes', default=None, 
+    help='Number of PCA modes to be removed in the planet mapmaker.', type=int) 
+
+    args = parser.parse_args()
```

Comment:
In the present version, one can't analyze data unless the telescope/band are listed in the initial params file.

It would be nice to be able to specify those initial params (really just beam size and detector response?) on the command line, for cases when the params file won't know what I'm talking about (e.g. running on ACT data).